apps/io_tester/io_tester.cc

```diff
@@ -657,6 +665,99 @@ class write_io_class_data : public io_class_data {
     }
 };
 
+class unlink_class_data : public class_data {
+private:
+    sstring _dir_path{};
+    uint64_t _file_id_to_remove{0u};
+
+public:
+    unlink_class_data(job_config cfg) : class_data(std::move(cfg)) {
+        if (!_config.files_count.has_value()) {
+            throw std::runtime_error("request_type::unlink requires specifying 'files_count'");
+        }
+    }
+
+    future<> do_start(sstring path, directory_entry_type type) override {
+        if (type == directory_entry_type::directory) {
+            return do_start_on_directory(path);
+        }
+        throw std::runtime_error(format("Unsupported storage. {} should be directory", path));
+    }
+
+    future<size_t> issue_request(char *buf, io_intent* intent) override {
+        if (all_files_removed()) {
+            return make_ready_future<size_t>(1u);
```

Comment:
That's not really nice. The result would be silently skewed if all files are removed, but other workload continues to run. It should at least print a warning here, so that test runner knows that something dodn't go the way it was expected to.

Also, what if returning 0 here?

Reply:
The following new warning has been added:

```
fmt::print("[WARNING]: Cannot fulfill issue in unlink_class_data! All files have been removed for shard_id={}\n"
           "[WARNING]: Please create more files or adjust the frequency of unlinks.", this_shard_id());
```

Regarding returning `1u` here: I followed the approach from `cpu_class_data` that also does not perform any read or write operations.

In general, the returned value is accumulated in `class_data::_data` and is used only when throughput is calculated and emitted. However, in `unlink_class_data` throughput is not emitted. Therefore, usage of 0 is possible and it does not cause any difference in behavior of `unlink_class_data`.

Reply:
Actually, reporting back accumulated "throughput" or "IOPS" for unlink class would also be nice -- we'll see if regular IO running in parallel affects unlinkage rate or not

Reply:
Okay. The following results have been added to `unlink_class_data`:
- IOPS
- latencies: [average, max]
- stats: [total_requests]

I am not sure how to calculate the bandwidth - shall `unlink_class_data` report the size of file that was removed or maybe we would like to report `1u` like `cpu_data_class`?